hazelcast/src/test/java/com/hazelcast/cache/jsr/JsrTestUtil.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache.jsr;
+
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.instance.HazelcastInstanceFactory;
+
+import javax.cache.Caching;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.junit.Assert.fail;
+
+/**
+ * Utility class responsible for setup/cleanup of JSR member tests.
+ */
+public final class JsrTestUtil {
+
+    /**
+     * Keeps track of system properties set by this utility.
+     * <p>
+     * We have to manage the System properties by ourselves, since they are set in {@link org.junit.BeforeClass} methods,
+     * which are invoked before our Hazelcast {@link org.junit.runner.Runner} classes are copying the System properties
+     * to restore them for us.
+     */
+    private static final List<String> SYSTEM_PROPERTY_REGISTRY = new LinkedList<String>();
+
+    private JsrTestUtil() {
+    }
+
+    public static void setup() {
```

Comment:
The system properties become shared state. Could it be that these tests are going to run into issues because of this? So one tests sets it to some value, but before it complete, another test sets it to another value? If tests are run serially, this isn't an issue, but we are running many tests in parallel.

Reply:
The JSR tests always have been serial tests (on class and method level, so `HazelcastSerialClassRunner` and no `ParallelTest` annotation). The fixed issue is not concurrent execution, just the leftover of the `CachingProvider` and a following JSR test in the same JVM.

Reply:
I created a PR with a new helper method to check for the `ParallelTest` category. This may not be a candidate for the backport, but at least in the master we can add that check in the `JsrTestUtil`, so it cannot be used in parallel tests: https://github.com/hazelcast/hazelcast/pull/11295